fre/app/freapp.py

```diff
@@ -7,70 +7,118 @@
+              required = True)
+@click.option("-b", "--begin",
+              type = str,
+              help = "`begin` / `begin` (env var) ISO8601 datetime format specification for" + \
+                     " starting date of data, part of input target file name",
+              required = True)
+@click.option("-t", "--tmp_dir",
+              type = str,
+              help = "`TMPDIR` / `tmp_dir` (env var) temp directory for location of file " + \
+                     "read/writes",
+              required = True)
+@click.option("-r", "--remap_dir",
+              type = str,
+              help = "`fregridRemapDir` / `remap_dir` (env var) directory containing remap file" + \
+                     " for regridding",
+              required = True)
+@click.option("-s", "--source",
+              type = str,
+              help = "`source` / `source` (env var) source name for input target file name " + \
+                     "within input directory to target for regridding. the value for `source` " + \
+                     "must be present in at least one component's configuration fields",
+              required = True)
+@click.option("-g", "--grid_spec",
+              type = str,
+              help = "`gridSpec` / `grid_spec` (env var) file containing mosaic for regridding",
+              required = True)
+@click.option("-I", "--def_xy_interp",
+              type = str,
+              help = "`defaultxyInterp` / `def_xy_interp` (env var) default lat/lon resolution " + \
+                     "for output regridding. (change me? TODO)",
+              required = True)
+@click.pass_context
+def regrid(context,
+              input_dir, output_dir, begin, tmp_dir,
+              remap_dir, source, grid_spec, def_xy_interp ):
+    ''' regrid target netcdf file '''
+    context.forward(_regrid_xy)
 
 @app_cli.command()
 @click.option("-i", "--infile",
-              type=str,
-              help="Input NetCDF file containing pressure-level output to be masked",
-              required=True)
+              type = str,
+              help = "Input NetCDF file containing pressure-level output to be masked",
+              required = True)
 @click.option("-o", "--outfile",
-              type=str,
-              help="Output file",
-              required=True)
+              type = str,
+              help = "Output file",
+              required = True)
 @click.option("-p", "--psfile",
-              help="Input NetCDF file containing surface pressure (ps)",
-              required=True)
+              help = "Input NetCDF file containing surface pressure (ps)",
+              required = True)
 @click.pass_context
 def mask_atmos_plevel(context, infile, outfile, psfile):
-    # pylint: disable=unused-argument
+    # pylint: disable = unused-argument
     """Mask out pressure level diagnostic output below land surface"""
     context.forward(mask_atmos_plevel_subtool)
 
 
 @app_cli.command()
 @click.option("-i", "--inf",
```

Comment:
Maybe, to keep consistent, also make `--inf` --> `--infile`

Reply:
that's `gen-time-averages` and would entail going through to change the args. if you don't mind, i'll leave that for now!